contracts/src/v0.8/functions/dev/v1_X/FunctionsBilling.sol

```diff
@@ -159,7 +161,10 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
     uint72 donFee,
     uint72 adminFee
   ) internal view returns (uint96) {
-    uint256 executionGas = s_config.gasOverheadBeforeCallback + s_config.gasOverheadAfterCallback + callbackGasLimit;
+    uint256 executionGas = s_config.gasOverheadBeforeCallback +
+      s_config.gasOverheadAfterCallback +
+      callbackGasLimit +
+      ChainSpecificUtil._getCurrentTxL1GasFees(msg.data);
```

Comment:
This is not correct according to OP docs. The L1 data fee is the total fee that will be added to the transaction cost and not "gas used". We can't add it to execution gas and then multiply by gas price. As a sanity check I looked up current value from OP Oracle contract and it was 15,818,209,764,247 - way too high to make sense in this equation.

Am I wrong?

Reply:
Also a relevant snippet from VRF, which confirms my comment: https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol#L508-L512